src/alloc.c

```diff
@@ -308,6 +308,22 @@ DLLEXPORT jl_value_t *jl_new_struct_uninit(jl_datatype_t *type)
     return jv;
 }
 
+DLLEXPORT jl_bytevec_struct_t jl_bytevec(const uint8_t *data, size_t n)
```

Comment:
moving the code to that creates the jl_value_t tag would seem to make sense to me